File: logic/logic.py
def exercise():
    arr = [1, 2, 3, 4, 5]
    n = len(arr)
    for i in range(n // 2):
        arr[i], arr[n - i - 1] = arr[n - i - 1], arr[i]
    print(" ".join(map(str, arr)))

def exerciseII(s):
    left, right = 0, len(s) - 1
    while left < right:
        while left < right and not s[left].isalnum():
            left += 1
        while left < right and not s[right].isalnum():
            right -= 1
        if s[left].lower() != s[right].lower():
            return False
        left += 1
        right -= 1
    return True

def exerciseIII(arr, target):
    seen = set()
    pairs = set()
    for num in arr:
        comp = target - num
        if comp in seen:
            pairs.add((min(num, comp), max(num, comp)))
        seen.add(num)
    if not pairs:
        print("No pairs found.")
    else:
        print(f"Pairs: {pairs}")

File: logic/logic.js
function exerciseI() {
    const arr = [1, 2, 3, 4, 5];
    const n = arr.length;
  
    for (let i = 0; i < Math.floor(n / 2); i++) {
      let temp = arr[i];
      arr[i] = arr[n - i - 1];
      arr[n - i - 1] = temp;
    }
  
    console.log(arr.join(" "));
  }
  
  function exerciseII(s) {
    let left = 0, right = s.length - 1;
  
    while (left < right) {
      while (left < right && !/[a-z0-9]/i.test(s[left])) left++;
      while (left < right && !/[a-z0-9]/i.test(s[right])) right--;
      if (s[left].toLowerCase() !== s[right].toLowerCase()) return false;
      left++;
      right--;
    }
  
    return true;
  }
  
  function exerciseIII(arr, target) {
    const seen = new Set();
    const pairs = new Set();
  
    for (let num of arr) {
      const complement = target - num;
      if (seen.has(complement)) {
        const pair = [Math.min(num, complement), Math.max(num, complement)].toString();
        pairs.add(pair);
      }
      seen.add(num);
    }
  
    if (pairs.size === 0) console.log("No pairs found");
    else console.log("Pairs:", [...pairs]);
  }
  
